components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java

```diff
@@ -76,6 +74,11 @@ public void startRoute(String id) throws Exception {
         camelContext.getRouteController().startRoute(id);
     }
 
+    public void resetRoute(String id) throws Exception {
+        ManagedRouteMBean resetRoute = camelContext.getManagedRoute(id, ManagedRouteMBean.class);
+        resetRoute.reset();
```

Comment:
Yeah I think an exception is overkill. There must be some more graceful HTTP Status 204 NO_CONTENT or maybe a 404 or something to return instead.